src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -2127,79 +2166,1057 @@ private void doAlignAssignment(int startRow,
    @Handler
    void onReformatCode()
    {
-      // Only allow if entire selection in R or C++ mode for now
-      if (!(DocumentMode.isSelectionInCppMode(docDisplay_) ||
-            DocumentMode.isSelectionInRMode(docDisplay_)))
+      // Only allow if entire selection in R mode for now
+      if (!DocumentMode.isSelectionInRMode(docDisplay_))
```

Comment:
I think if you use this command when not in R mode we should display an explicit warning (there is a re-usable showRModeWarning function for this which you can see called from onExtractLocalVariable).